Sources/xcproj/PBXProj.swift

```diff
@@ -3,8 +3,135 @@ import PathKit
 
 /// Represents a .pbxproj file
 public class PBXProj: Decodable {
+    public class Objects: Equatable {
+        // MARK: - Properties
+        public var buildFiles: ReferenceableCollection<PBXBuildFile> = ReferenceableCollection<PBXBuildFile>()
+        public var aggregateTargets: ReferenceableCollection<PBXAggregateTarget> = ReferenceableCollection<PBXAggregateTarget>()
+        public var containerItemProxies: ReferenceableCollection<PBXContainerItemProxy> = ReferenceableCollection<PBXContainerItemProxy>()
+        public var groups: ReferenceableCollection<PBXGroup> = ReferenceableCollection<PBXGroup>()
+        public var fileElements: ReferenceableCollection<PBXFileElement> = ReferenceableCollection<PBXFileElement>()
+        public var configurationLists: ReferenceableCollection<XCConfigurationList> = ReferenceableCollection<XCConfigurationList>()
+        public var versionGroups: ReferenceableCollection<XCVersionGroup> = ReferenceableCollection<XCVersionGroup>()
+        public var buildConfigurations: ReferenceableCollection<XCBuildConfiguration> = ReferenceableCollection<XCBuildConfiguration>()
+        public var variantGroups: ReferenceableCollection<PBXVariantGroup> = ReferenceableCollection<PBXVariantGroup>()
+        public var targetDependencies: ReferenceableCollection<PBXTargetDependency> = ReferenceableCollection<PBXTargetDependency>()
+
+        public var nativeTargets: ReferenceableCollection<PBXNativeTarget> = ReferenceableCollection<PBXNativeTarget>()
+        public var fileReferences: ReferenceableCollection<PBXFileReference> = ReferenceableCollection<PBXFileReference>()
+        public var projects: ReferenceableCollection<PBXProject> = ReferenceableCollection<PBXProject>()
+        public var referenceProxies: ReferenceableCollection<PBXReferenceProxy> = ReferenceableCollection<PBXReferenceProxy>()
+
+
+        // Ordering matters with build phases so leave these as arrays
```

Comment:
@rahul-malik I think I'm missing something here. Where does the order matter?

Reply:
Yeah these can also be ReferenceableCollection